Ignore default security context constraints when running on kube

derekwaynecarr opened the pull request:
If you are running OpenShift on a Kubernetes cluster, we need to ignore security context constraints since they are not actually in upstream Kubernetes...

derekwaynecarr commented:
/cc @liggitt per our chat earlier...

deads2k commented:
Is it too painful to make this decision based on discovery information from the kube server?  That seems more correct than basing on whether or not we're running an embedded kube.

liggitt commented:
this really just ignores errors we tolerate... is it worth it?

paralin commented:
I noticed these errors when running openshift on an existing kubernetes server. So +1 for merging something similar to this.

liggitt commented:
To be clear, it is only logging, it shouldn't affect anything functionally

deads2k commented:
> this really just ignores errors we tolerate... is it worth it?

Depends on the pain.  I don't feel strongly for a one-off, but I worry about this becoming something like the atomic switch we had that rippled through for builds.

derekwaynecarr commented:
* Fixed the gofmt issue (embarrasing that VS Code was not gofmt-ing origin yet).
* Added a TODO to move to discovery client.

As for the atomic switch comparison, thus far this is the only change I have identified where running OpenShift on Kubernetes can cause confusing log output.  Of course, some features just don't work, but for the experimental command we are trying to put together, I think it's fine.

derekwaynecarr commented:
eh, thinking about it should not be too hard to switch to discovery client.  will try that now.

derekwaynecarr commented:
@deads2k @liggitt - now with discovery!  PTAL

deads2k commented on pkg/cmd/server/origin/ensure.go:
```diff
@@ -151,7 +153,47 @@ func (c *MasterConfig) ensureNamespaceServiceAccountRoleBindings(namespace *kapi
 	}
 }
 
+func (c *MasterConfig) securityContextConstraintsSupported() (bool, error) {
```
Add TODO to make this a library upstream, `ResourceExists(GroupVersionResource)` or some such.  Definitely not a blocker, just seen a few these ripple through.

deads2k commented:
lgtm.

derekwaynecarr commented:
Any objections to merging?

paralin commented:
Seems good to me, I merged this into my working tree a few days ago, no problems

deads2k commented:
[merge]

bparees commented:
[merge] vagrant plugin was broken.
